x-pack/plugins/canvas/canvas_plugin_src/functions/common/math.js

```diff
@@ -5,54 +5,57 @@
  */
 
 import { evaluate } from 'tinymath';
+import { getFunctionErrors } from '../../errors';
 import { pivotObjectArray } from '../../../common/lib/pivot_object_array';
 
-export const math = () => ({
-  name: 'math',
-  type: 'number',
-  help:
-    'Interpret a math expression, with a number or datatable as context. Datatable columns are available by their column name. ' +
-    'If you pass in a number it is available as "value" (without the quotes)',
-  context: {
-    types: ['number', 'datatable'],
-  },
-  args: {
-    expression: {
-      aliases: ['_'],
-      types: ['string'],
-      help:
-        'An evaluated TinyMath expression. (See [TinyMath Functions](https://www.elastic.co/guide/en/kibana/current/canvas-tinymath-functions.html))',
+export const math = () => {
+  const functionErrors = getFunctionErrors();
+
+  return {
+    name: 'math',
+    type: 'number',
+    help:
+      'Interpret a math expression, with a number or datatable as context. Datatable columns are available by their column name. ' +
+      'If you pass in a number it is available as "value" (without the quotes)',
+    context: {
+      types: ['number', 'datatable'],
+    },
+    args: {
+      expression: {
+        aliases: ['_'],
+        types: ['string'],
+        help:
+          'An evaluated TinyMath expression. (See [TinyMath Functions](https://www.elastic.co/guide/en/kibana/current/canvas-tinymath-functions.html))',
+      },
     },
-  },
-  fn: (context, args) => {
-    if (!args.expression || args.expression.trim() === '') {
-      throw new Error('Empty expression');
-    }
+    fn: (context, args) => {
+      if (!args.expression || args.expression.trim() === '') {
+        throw functionErrors.math.expressionEmpty();
+      }
 
-    const isDatatable = context && context.type === 'datatable';
-    const mathContext = isDatatable
-      ? pivotObjectArray(context.rows, context.columns.map(col => col.name))
-      : { value: context };
-    try {
-      const result = evaluate(args.expression, mathContext);
-      if (Array.isArray(result)) {
-        if (result.length === 1) {
-          return result[0];
+      const isDatatable = context && context.type === 'datatable';
+      const mathContext = isDatatable
+        ? pivotObjectArray(context.rows, context.columns.map(col => col.name))
+        : { value: context };
+      try {
+        const result = evaluate(args.expression, mathContext);
+        if (Array.isArray(result)) {
+          if (result.length === 1) {
+            return result[0];
+          }
+          throw functionErrors.math.resultLengthInvalid();
+        }
+        if (isNaN(result)) {
+          throw functionErrors.math.executeFailure();
+        }
+        return result;
+      } catch (e) {
+        if (context.rows.length === 0) {
+          throw functionErrors.math.datatableEmpty();
+        } else {
+          throw e;
```

Comment:
fyi, this can result in non-translated errors. It's more of an upstream issue, but there's some cases we could test for and translate too. One example is using an invalid column name:

![screenshot 2019-01-15 13 38 52](https://user-images.githubusercontent.com/404731/51208439-ea9b0880-18ca-11e9-8832-5a07a65c9511.png)